src/main/java/ru/hogwarts/school/service/StudentService.java

```diff
@@ -1,44 +1,74 @@
 package ru.hogwarts.school.service;
 
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repositories.StudentRepository;
+
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
-import ru.hogwarts.school.model.Student;
+import java.util.List;
+
+import static java.lang.reflect.Array.get;
+
+@Service
+@AllArgsConstructor
+public class StudentService {
+    @Autowired
+    private final StudentRepository studentRepository;
+
+  
+
+    public Student findStudent(long id) {
+        return studentRepository.getById(id);
+    }
 
-    public class StudentService {
-
-        private final HashMap<Long, Student> students = new HashMap<>();
-        private long count = 0;
-
-        public Student addStudent(Student student) {
-            student.setId(count++);
-            students.put(student.getId(), student);
-            return student;
-        }
-
-        public Student findStudent(long id) {
-            return students.get(id);
-        }
-
-        public Student editStudent(Student student) {
-            if (!students.containsKey(student.getId())) {
-                return null;
-            }
-            students.put(student.getId(), student);
-            return student;
-        }
-
-        public Student deleteStudent(long id) {
-            return students.remove(id);
-        }
-
-        public Collection<Student> findByAge(int age) {
-            ArrayList<Student> result = new ArrayList<>();
-            for (Student student : students.values()) {
-                if (student.getAge() == age) {
-                    result.add(student);
-                }
-            }
-            return result;
-        }
+    public Student editStudent(Student student) {
+       return studentRepository.save(student);
+    }
+
+    public void deleteStudent(long id) {
+        studentRepository.deleteById(id);
+    }
+    public Collection<Student> findByAge(int age) {
+
+        return studentRepository.findAllByAge(age);
+}
+    public Student get(Long id) {
+        return studentRepository.findById(id).orElse(null);
+    }
+
+    public Student addStudent(Student student) {
+        return studentRepository.save(student);
+    }
+
+
+    public List<Student> findByFacultyId(Long facultyId) {
+        return studentRepository.findByFacultyId(facultyId);
+    }
+
+    public Collection<Student> getByAge(int age) {
+        return studentRepository.findAllByAge(age);
+    }
+
+
+
+
+    public Collection<Student> findAll() {
+        return studentRepository.findAll();
+    }
+
+    public List<Student> findByAgeBetween(int min, int max) {
+        return StudentRepository.findByAgeBetween(min, max);
```

Comment:
И тут, соответственно, обращаемся к методу через объект
```suggestion
        return studentRepository.findByAgeBetween(min, max);
```

Reply:
опечатка)